src/helpers/MiscFunctions.cpp

```diff
@@ -330,8 +331,28 @@ int getWorkspaceIDFromString(const std::string& in, std::string& outName) {
 
             std::sort(validWSes.begin(), validWSes.end());
 
+            if (includeEmpty && validWSes.size() >= 2) {
+                // Include all empty workspaces in between the known workspaces
+                // We cannot just add all empty workspaces beforehand, since then validWSes would be potentially of size 2^31-1
+                for (auto it = validWSes.begin(); it != validWSes.end() - 1; ++it) {
+                    // add all empty workspaces in between current and last
+                    int curValidWSID  = *it;
+                    int nextValidWSID = *(it + 1);
+                    // ]curValidWSID, nextValidWSID[
+                    for (int id = curValidWSID + 1; id < nextValidWSID; id++) {
```

Comment:
issue: if someone opens workspace id `3483267` this will literally hang for a couple seconds

Reply:
Good catch. I was thinking of only searching up to |remains| empty workspaces in the direction we're looking in. This should prevent hangs (unless someone calls hyprctl dispatch r+30000 or so, but I don't think that is realistic)